docs/src/tutorials/point_process.md

```diff
@@ -0,0 +1,826 @@
+number of points (or events) in between time ``5`` and ``10`` inclusive. The
+number of points in this interval is a random variable. If ``N`` is a Poisson
+process with conditional intensity (or rate) equal to ``1``, then ``N[5, 10]``
+is distributed according to a Poisson distribution with parameter ``\lambda = 5``.
+
+For convenience, we denote ``N(t) \equiv N[0, t)`` as the number of points since
+the start of time until ``t``, exclusive of ``t``. A TPP is _simple_, if only a
+single event can occur in any unit of time ``t``, that is, ``\Pr(N[t, t +
+\epsilon) > 2) = 0``. We can then define a differential of ``N``, ``dN``, which
+describes the change in ``N(t)`` over an infinitesimal amount of time.
+
+```math
+dN(t) = \begin{cases}
+  1 \text{ , if } N[t, t + \epsilon] = 1 \\
+  0 \text{ , if } N[t, t + \epsilon] = 0.
+\end{cases}
+```
+
+Therefore, we can use any TPP to define a stochastic differential equations
+(SDE) with discontinuous jumps.
+
+```math
+du = f(u,p,t)dt + g(u,p,t) dW(t) + h(u,p,t) dN(t)
+```
+
+In the jump literature, ``N`` is usually a Poisson process which allows for
+a number of convenient properties for jump modelling. See Björk [^2] and Hanson [^3]
+for more theoretical discussions of SDE with discontinuous jumps.
+
+We can start implementing our TPP interface in Julia by working with the
+following SDE with discontinuous jumps.
+
+```math
+du = dN(t)
+```
+
+In this case, ``u(t)`` is a monotonic function which counts the total number of
+points since the start of time. Therefore, there is a one-to-one map between
+``N`` and ``u``.
+
+A TPP is marked if, in addition to the temporal support ``\mathbb{R}``, there is
+a mark space ``\mathcal{K}`` such that ``N`` is a random integer measure over
+``\mathbb{R} \times \mathcal{K}``. Intuitively, for every point in the process
+there is a mark associated with it. If the mark space is discrete, we have
+a multivariate TPP. There are different ways to interpret TPPs, and we will
+move between these interpretations throughout the tutorial.
+
+To make the connection between the JumpProcesses library and point process
+theory, we will compare against the
+[PointProcess.jl](https://github.com/gdalle/PointProcesses.jl) library, which
+offers a common interface for marked TPPs. Since TPP sampling is more efficient
+if we split any marked TPP into a sparsely connected multivariate TPP, we define
+`SciMLPointProcess` as a multivariate TPP such that each sub-component is itself
+a marked TPP on a continuous space. Therefore, we have that our structure
+includes a vector of sub-TPP `jumps`, a vector of mark distributions `mark_dist`
+and the sparsely connected graph `g`.
+
+```@example tpp
+using JumpProcesses
+using PointProcesses
+
+struct SciMLPointProcess{M, J <: JumpProcesses.AbstractJump, G, D, T <: Real} <:
+       AbstractPointProcess{M}
+    jumps::Vector{J}
+    mark_dist::Vector{D}
+    g::G
+    p::Any
```

Comment:
@gzagatti why `Any`?

Reply:
No particular reason. I just meant to say the parameters are up to the user. Should we remove `Any` or use `Tuple` instead?

Reply:
In Julia, I never know when to use `Any` and when to leave it blank.